crates/cxx-qt-build/src/lib.rs

```diff
@@ -730,6 +751,35 @@ impl CxxQtBuilder {
 
             let mut qml_metatypes_json = Vec::new();
 
+            // Check that all rust files are within the same directory
+            //
+            // Note we need to do this as moc generates an inputFile which only
+            // includes the file name, qmltyperegistrar then uses this for the
+            // include path (and doesn't consider any prefix).
+            //
+            // This can also be observed when using qt_add_qml_module, if a class
+            // has a QML_ELEMENT the file must be in the same directory as the
+            // CMakeLists and cannot be a relative path to a sub directory.
+            let dirs = qml_module
+                .rust_files
+                .iter()
+                .map(|file| {
+                    file.parent()
+                        .unwrap()
+                        .file_name()
+                        .unwrap()
```

Comment:
I think this is incorrect.

If we compare the file_name of the parent, we'll only be comparing the very last directory name.
So e.g. if we have this as rust_files:
```
src/generator/rust/property/mod.rs
src/generator/cpp/property/mod.rs
```
This will find `property` and `property` and discard the discrepancy in the earlier paths.

Reply:
ah yes it should be the whole of `parent()` not just the filename